modules/apps/commerce/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/shared/search/CPSearchResultsPortletSharedSearchContributor.java

```diff
@@ -179,41 +175,38 @@ private void _paginate(
 		portletSharedSearchSettings.setPaginationStartParameterName(
 			paginationStartParameterName);
 
-		Optional<String> paginationStartParameterValueOptional =
-			portletSharedSearchSettings.getParameterOptional(
+		String paginationStartParameterValue =
+			portletSharedSearchSettings.getParameter(
 				paginationStartParameterName);
 
-		Optional<Integer> paginationStartOptional =
-			paginationStartParameterValueOptional.map(Integer::valueOf);
-
-		paginationStartOptional.ifPresent(
-			portletSharedSearchSettings::setPaginationStart);
+		if (paginationStartParameterValue != null) {
+			portletSharedSearchSettings.setPaginationStart(
+				GetterUtil.getInteger(paginationStartParameterValue));
```

Comment:
@julschong Originally the logic uses `Integer::valueOf` to convert the string to integer, which throws `NumberFormatException` when the string can not be parsed into an integer.

`GetterUtil#getInteger`, however, returns `0` when the string can't be parsed. This is yet another inconsistency in logic before/after the conversion.

Let's always keep the logic consistent in the conversion commits, and, if you think it's necessary, add extra commits to refactor.